numpy/core/src/multiarray/compiled_base.c

```diff
@@ -80,6 +80,357 @@ minmax(const npy_intp *data, npy_intp data_len, npy_intp *mn, npy_intp *mx)
+        hist = (PyArrayObject *)PyArray_ZEROS(1, &bins, NPY_INTP, 0);
+        if (hist == NULL) {
+            goto fail;
+        }
+        Py_DECREF(arr_edges);
+        Py_DECREF(arr_x);
+        return (PyObject *)hist;
+    }
+
+    x = (double *)PyArray_DATA(arr_x);
+    e = (double *)PyArray_DATA(arr_edges);
+
+    min = e[0];
+    max = e[bins];
+
+    if (bins <= 0 || (max - min) <= 0)
+    {
+        PyErr_SetString(PyExc_ValueError, "Bin edges must be increasing");
+        goto fail;
+    }
+    norm = bins / (max - min);
+
+    if (weights == Py_None) {
+        npy_intp i, *ihist;
+
+        hist = (PyArrayObject *)PyArray_ZEROS(1, &bins, NPY_INTP, 0);
+        if (hist == NULL) {
+            goto fail;
+        }
+
+        ihist = (npy_intp *)PyArray_DATA(hist);
+
+        NPY_BEGIN_ALLOW_THREADS;
+        for(i = 0; i < len; i++) {
+            double tx = x[i];
+            if ((min <= tx) && (tx <= max)) {
+                npy_intp ix = (npy_intp)((tx - min) * norm);
+                /*
+                 * For values that lie exactly on mx we need to subtract one.
+                 * Also, the index computation is not guaranteed to give
+                 * exactly consistent results within ~1 ULP of the bin edges.
+                 * The last bin includes the right edge. The other bins do not.
+                 */
+                if (ix == bins || (ix != 0 && tx < e[ix])) {
+                    ix--;
+                }
+                else if ((ix + 1) != bins && tx >= e[ix + 1]) {
+                    ix++;
+                }
+                ihist[ix]++;
+            }
+        }
+        NPY_END_ALLOW_THREADS;
+    }
+    else {
+        npy_intp i;
+        double *w, *dhist;
+
+        arr_w = (PyArrayObject *)PyArray_ContiguousFromAny(weights, NPY_DOUBLE, 1, 1);
+        if (arr_w == NULL) {
+            goto fail;
+        }
+        if (PyArray_SIZE(arr_w) != len) {
+            PyErr_SetString(PyExc_ValueError,
+                            "Weights must be same length as array");
+            goto fail;
+        }
+
+        hist = (PyArrayObject *)PyArray_ZEROS(1, &bins, NPY_DOUBLE, 0);
+        if (hist == NULL) {
+            goto fail;
+        }
+
+        w = (double *)PyArray_DATA(arr_w);
+        dhist = (double *)PyArray_DATA(hist);
+
+        NPY_BEGIN_ALLOW_THREADS;
+        for(i = 0; i < len; i++) {
+            double tx = x[i];
+            if ((min <= tx) && (tx <= max)) {
+                npy_intp ix = (npy_intp)((tx - min) * norm);
+                if (ix == bins || (ix != 0 && tx < e[ix])) {
+                    ix--;
+                }
+                else if ((ix + 1) != bins && tx >= e[ix + 1]) {
+                    ix++;
+                }
+                dhist[ix] += w[i];
+            }
+        }
+        NPY_END_ALLOW_THREADS;
+
+        Py_DECREF(arr_w);
+    }
+
+    Py_DECREF(arr_edges);
+    Py_DECREF(arr_x);
+    return (PyObject *)hist;
+
+fail:
+    Py_XDECREF(hist);
```

Comment:
This line will never decref - but maybe it's better to leave it here in case that changes in future

Reply:
good catch. and I agree leaving it in does no harm and might do good later.